libraries/AP_EFI/AP_EFI_Serial_Hirth.h

```diff
@@ -0,0 +1,209 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+
+#pragma once
+
+#include "AP_EFI_config.h"
+
+#if AP_EFI_SERIAL_HIRTH_ENABLED
+#include "AP_EFI.h"
+#include "AP_EFI_Backend.h"
+
+#define BYTE_RANGE_MAX 256
+
+/*!
+ * Data mapping between rawBytes and Telemetry packets
+ */
+typedef struct {
+    uint8_t quantity;
+    uint8_t code;
+    uint8_t checksum;
+} data_set_t;
+
+
+/*!
+ * class definition for Hirth 4103 ECU
+ */
+class AP_EFI_Serial_Hirth: public AP_EFI_Backend {
+public:
+    AP_EFI_Serial_Hirth(AP_EFI &_frontend);
+
+    void update() override;
+
+    void check_response();
+
+    void send_request();
+
+    void decode_data();
+
+    bool send_request_status();
+
+    bool send_target_values(uint16_t);
+
+    void get_quantity();
+
+    void log_status();
+
+private:
+    // serial port instance
+    AP_HAL::UARTDriver *port;
+
+    // periodic refresh 
+    uint32_t last_response_ms;
+    uint32_t last_packet_ms;
+    uint32_t last_req_send_throttle_ms;
+
+    // raw bytes - max size
+    uint8_t raw_data[BYTE_RANGE_MAX];
+
+    // request and response data 
+    data_set_t req_data;
```

Comment:
```suggestion
    uint8_t requested_code;
```

... saves a bunch of pointless assignments